src/Campaigns/Routes/Campaign.php

```diff
@@ -0,0 +1,190 @@
+use WP_REST_Server;
+
+/**
+ * @unreleased
+ */
+class Campaign implements RestRoute
+{
+
+    /** @var string */
+    protected $endpoint = 'campaigns/(?P<id>[0-9]+)';
+
+    /**
+     * @unreleased
+     */
+    public function registerRoute()
+    {
+        // Get campaign
+        register_rest_route(
+            'give-api/v2',
+            $this->endpoint,
+            [
+                [
+                    'methods' => WP_REST_Server::READABLE,
+                    'callback' => [$this, 'handleGetRequest'],
+                    'permission_callback' => function () {
+                        return current_user_can('manage_options');
+                    },
+                ],
+                'args' => [
+                    'id' => [
+                        'type' => 'integer',
+                        'required' => true,
+                        'validate_callback' => function ($id) {
+                            return filter_var($id, FILTER_VALIDATE_INT);
+                        },
+                    ],
+                ],
+            ]
+        );
+
+
+        // Update Campaign
+        register_rest_route(
+            'give-api/v2',
+            $this->endpoint,
+            [
+                [
+                    'methods' => WP_REST_Server::EDITABLE,
+                    'callback' => [$this, 'handleUpdateRequest'],
+                    'permission_callback' => function () {
+                        return current_user_can('manage_options');
+                    },
+                ],
+                'args' => [
+                    'id' => [
+                        'type' => 'integer',
+                        'required' => true,
+                        'validate_callback' => function ($id) {
+                            return filter_var($id, FILTER_VALIDATE_INT);
+                        },
+                    ],
+                    'title' => [
+                        'type' => 'string',
+                        'required' => false,
+                        'sanitize_callback' => 'sanitize_text_field',
+                    ],
+                ],
+                'schema' => [$this, 'getSchema'],
+            ]
+        );
+    }
+
+    /**
+     * @unreleased
+     *
+     * @throws Exception
+     */
+    public function handleGetRequest(WP_REST_Request $request): WP_Rest_Response
+    {
+        $campaign = CampaignModel::find($request->get_param('id'));
+
+        if ( ! $campaign) {
+            return new WP_REST_Response([
+                'message' => __('Campaign not found', 'give'),
+            ], 400);
+        }
+
+        return new WP_REST_Response($campaign->toArray());
+    }
+
+    /**
+     * @unreleased
+     *
+     * @throws Exception
+     */
+    public function handleUpdateRequest(WP_REST_Request $request): WP_REST_Response
+    {
+        $campaign = CampaignModel::find($request->get_param('id'));
+
+        if ( ! $campaign) {
+            return new WP_REST_Response([
```

Comment:
You'll want to return a WP_Error to properly format errors.

Reply:
Resolved: https://github.com/impress-org/givewp/pull/7547/commits/d57d213d7df34156fb89fd8cd904c3838b5e2de5